### PR TITLE
Master Table

### PR DESCRIPTION
**Pull request by MonishaSivanathan**
- Added master table for medication
- Added model, controller, router
- Handled Exception and Logger

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -49,6 +50,7 @@
 app.include_router(hl7_immunization_router.router, tags=["IMMUNIZATION"])
 app.include_router(authentication_router.router, tags=["AUTHENTICATION"])
 app.include_router(condition_allergy_routes.router, tags=["ALLERGY_CONDITION"])
+app.include_router(medication_list_routes.router, tags=["MEDICATION_LIST"])
```
Comment (tamsud) [inline main.py:53]: add a 2 routers.
1. FOR Medication list
2. FOR allergy list

File: routes/medication_list_routes.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+from controller.medication_list_controller import MedicationList
+
+router = APIRouter()
+logger = logging.getLogger("log")
+
+
+@router.get("/medication_list/{medication_name}")
+async def get_medication_list(medication_name: str):
+        logger.info(f"Medication Name:{medication_name}")
```
Comment (tamsud) [inline routes/medication_list_routes.py:13]: In router add a condition to check for the minimum two character. if the length is less than 2 char make it as a response None

File: controller/medication_list_controller.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+from aidbox.base import API
+
+logger = logging.getLogger("log")
+
+
+class MedicationList:
+    @staticmethod
+    def get_medication_list(medication_name: str):
+        try:
+            medication_list = API.make_request(
```
Comment (tamsud) [inline controller/medication_list_controller.py:15]: Add a logger for the response after making the AIDBIX api call

File: controller/medication_list_controller.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                method="GET",
+                endpoint=f"/Concept?.definition$contains={medication_name}&system=http://hl7.org/fhir/sid/icd-10",
+            )
+            medication_list_snomed = API.make_request(
```
Comment (tamsud) [inline controller/medication_list_controller.py:19]: Need allergy list in separate API

**Review by tamsud — CHANGES_REQUESTED** [review]
@MonishaSivanathan Check the given review comments and do the necessary changes

File: routes/medication_list_routes.py (hunk 2 above)
Comment (MonishaSivanathan) [reply routes/medication_list_routes.py]: i have updated

File: controller/medication_list_controller.py (hunk 3 above)
Comment (MonishaSivanathan) [reply controller/medication_list_controller.py]: added logger

**Review by tamsud — APPROVED** [review]

File: routes/master_table_routes.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        if len(medication_name) < 2:
+                logger.info("Medication name must be at least 2 characters long.")
+                raise HTTPException(status_code=400, detail="Medication name must be at least 2 characters long.")
+        logger.info(f"Medication Name:{medication_name}")
+        return MedicationList.get_medication_list(medication_name)
+
+@router.get("/allergy_list/{allergy_name}")
+async def get_allergy_list(allergy_name: str):
+        if len(allergy_name) < 2:
+                logger.info("Allergy name must be at least 2 characters long.")
```
[the remark this answers is not in the document]
Comment (MonishaSivanathan) [reply routes/master_table_routes.py]: added

File: routes/master_table_routes.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+from controller.medication_list_controller import MedicationList
+
+router = APIRouter()
+logger = logging.getLogger("log")
+
+
+@router.get("/medication_list/{medication_name}")
+async def get_medication_list(medication_name: str):
+        if len(medication_name) < 2:
```
[the remark this answers is not in the document]
Comment (MonishaSivanathan) [reply routes/master_table_routes.py]: i have updated

**Pull request merged**